app/PasswordStorage.php

```diff
@@ -7,7 +7,7 @@ class PasswordStorage
     static function set(string $pwd)
     {
         if (self::apcuAvailable()) {
-            $result = apcu_store('firefly_fints_bank_password', $pwd);
+            $result = apcu_store('firefly_fints_bank_password', $pwd, 600);
```

Comment:
I think such a ttl is a potential source for confusing behavior for the user. If he is interrupted while filling out the form and then continues, there is an error and no hint as to what could be the issue. 
Could you:
- increase this to 1h and
- add an explanation in the GUI. Something like 
```
{% if apcu_enabled %}
Please note that your bank password will be automatically cleared from memory after 1 hour.
{% endif %}
```

Reply:
I simply removed the timeout for now, as it is only in memory. I'm pretty sure this PR would also have fixed #141 but it seems it did not fix #138. However, i tried multiple code paths on a Windows+Docker Setup as well as a Baremetal+Linux+Nginx and can not reproduce the issue. Therefore, i'm not sure how to proceed. As you have seen, i'm not hesitant to put in the work, but i need some input on how to proceed? (Or if you want to drop this feature and be done with it)

Reply:
Hey @TyrionWarMage 

1) I didn't realize this PR contained a fix for #141.
If you like, you can resubmit the feature of #120 with this fix included as a new PR :)

2) Regarding #138: I'm also not 100% sure how to proceed here. According to @Benniphx [here](https://github.com/bnw/firefly-iii-fints-importer/issues/138#issuecomment-2301822344), the issue is not resolved with the changes you suggested. If @Benniphx (or one of the other persons affected by #138) is willing to test some more, maybe you could provide him with a custom branch which has more debug output and logging. If @Benniphx runs that and gives you the logs, maybe you could figure out where the issue lies? 
   I like the feature you added. I'd be happy to keep it, providing the bug is fixed :)

Reply:
1. I assume this PR would have resolved things because the faulty cleanup step included a session cleanup and #141 sounds like a premature session cleanup.
2. In my tests, i was able to indirectly reproduce his issues with triggering the device selection branch, which also resulted in the premature cleanup and therefore the password not being available.

In conclusion, from my point of view, this PR is fine now (i also use this in my production setup). However, i think before we merge this again, someone else (you? 😄 ) should also test this, because the problems are in relation to a code path that i am likely not using.